lightningd/options.c

```diff
@@ -193,6 +183,34 @@ static char *opt_add_announce_addr(const char *arg, struct lightningd *ld)
 	return NULL;
 }
 
+static char *opt_add_addr(const char *arg, struct lightningd *ld)
+{
+	/* handle in case you used the addr option with an .onion */
+	if (strends(arg, ".onion")) {
+			log_unusual(ld->log, "You used  ** --addr=%s ** option with an .onion address, please use"
+					" --announce-addr ! You are lucky in this node live some wizards and"
+					" fairies, we have done this for you and announce, Be as hidden as wished"
+					, arg);
+			return opt_add_announce_addr(arg, ld);
+	}
+	/* the intended call */
+	return opt_add_addr_withtype(arg, ld, ADDR_LISTEN_AND_ANNOUNCE, true);
+}
+
+static char *opt_add_bind_addr(const char *arg, struct lightningd *ld)
+{
+	/* handle in case you used the bind option with an .onion */
+	if (strends(arg, ".onion")) {
+			log_unusual(ld->log, "You used  ** --addr-bind=%s ** option with an .onion address,"
```

Comment:
```suggestion
			log_unusual(ld->log, "You used `--addr-bind=%s` option with an .onion address,"
```

Reply:
done :+1: And changed `if (strends(arg, ".onion")) {` to a more  bolder check to allow xy.onion:[port]. 

siggh: The first commit with just `return` false; in connectd.c was not so long but
          @rustyrussell idea with "I think you want ..." resonated, so ...
and now I have found at other places also some new issues that need extra PR, please nack so I can go back in my mirror :smile: and tell you only what you wish for ...